crypto.c

```diff
@@ -387,38 +387,33 @@ static void dtls_ec_key_from_uint32(const uint32_t *key, size_t key_size,
 
 int dtls_ec_key_from_uint32_asn1(const uint32_t *key, size_t key_size,
 				 unsigned char *buf) {
-  int i;
-  unsigned char *buf_orig = buf;
-  int first = 1; 
-
-  for (i = (key_size / sizeof(uint32_t)) - 1; i >= 0 ; i--) {
-    if (key[i] == 0)
-      continue;
-    /* the first bit has to be set to zero, to indicate a poritive integer */
-    if (first && key[i] & 0x80000000) {
-      *buf = 0;
-      buf++;
-      dtls_int_to_uint32(buf, key[i]);
-      buf += 4;      
-    } else if (first && !(key[i] & 0xFF800000)) {
-      buf[0] = (key[i] >> 16) & 0xff;
-      buf[1] = (key[i] >> 8) & 0xff;
-      buf[2] = key[i] & 0xff;
-      buf += 3;
-    } else if (first && !(key[i] & 0xFFFF8000)) {
-      buf[0] = (key[i] >> 8) & 0xff;
-      buf[1] = key[i] & 0xff;
-      buf += 2;
-    } else if (first && !(key[i] & 0xFFFFFF80)) {
-      buf[0] = key[i] & 0xff;
-      buf += 1;
-    } else {
-      dtls_int_to_uint32(buf, key[i]);
-      buf += 4;
-    }
-    first = 0;
+  int i = 0;
+   
+  dtls_ec_key_from_uint32(key, key_size, buf);
+  
+  /* skip leading 0's */
+  while (i < (int)key_size && buf[i] == 0) {
+     ++i;
+  }
+  if (i == (int)key_size) {
+      /* do some error reporting */
```

Comment:
the comment was a "placeholder", though I'm not common, how such things are reported. Maybe using `dtls_alert("ecc key only contains 0s\n");`?

Reply:
Thanks for spotting this - corrected code pushed.